crates/web5_cli/src/dids/create.rs

```diff
@@ -75,27 +90,55 @@ impl Commands {
             }
             Commands::Web {
                 domain,
+                service_endpoint,
+                service_endpoint_type,
                 no_indent,
                 json_escape,
             } => {
                 // Check if the current process has root privileges because the InMemoryKeyManager may require root privileges
                 warn_if_not_root();
                 let key_manager = Arc::new(InMemoryKeyManager::new());
 
-                let bearer_did = DidWeb::create(
-                    domain,
-                    Some(DidWebCreateOptions {
-                        key_manager: Some(key_manager.clone()),
-                        ..Default::default()
-                    }),
-                )
-                .unwrap();
+                let mut did_web_create_options = DidWebCreateOptions {
+                    key_manager: Some(key_manager.clone()),
+                    ..Default::default()
+                };
+
+                // Parse the domain to extract the host without the protocol
+                let domain_host = match Url::parse(domain) {
```

Comment:
I believe all of this domain validation already occurs internal to the `DidWeb::create()` function https://github.com/TBD54566975/web5-rs/blob/6f6c336744ce0a35c5dc73e517be14f91c49b3b2/crates/web5/src/dids/methods/did_web/mod.rs#L92

Reply:
yup it does, but for the service endpoint id it needs to be a valid did web uri

Reply:
{
  "id": "did:web:www.banana.com#service-1",
  "type": "PFIService",
  "serviceEndpoint": [
    "https://example.com"
  ]
}